### DISCUSSION TITLE
Allow editor to add new fields to files collection

### DISCUSSION THREAD

**Discussion by Mahjouba91** [directus/directus#6404] [Feature Requests]
The files collection is a bit different from other collections as it's built in but at the end, it sits in the same database structure as any collection.

So I wonder why it's not possible to add a field directly from the UI or at least at the code level (from a contract like yaml file).

Because currently I only have two ways of doing it:

- Adding fields to a multiple files field (which mean it would not be available for a single image field)
- Adding new column directly to the files table in the DB, which unfortunately does not allow you to specify if it's a mandatory field or not, if it's a dropdown, to add a placeholder, etc. You can only have a free textfield.

Whereas my use case is pretty simple:

Add an alternative text field to images, + some other related fields like media source (dropdown with fixed values), attribution, etc.

What do you think? Is it like a bug, or just a lack of customisation that could be easily implemented in core?

**Comment by rijkvanzanten** [comment]
> What do you think? Is it like a bug, or just a lack of customisation that could be easily implemented in core?

None of the above 👍🏻 

It's already possible. You can go to Settings > Data Model, and enable "System Collections" in the right sidebar 👍🏻

**Reply by Mahjouba91** [reply]
Oh my, love that feature, thank you so much, is it documented somewhere?

**Reply by rijkvanzanten** [reply]
Good question! I'm not sure. Do you wanna open a PR to add a little note to https://docs.directus.io/guides/fields/ in https://github.com/directus/directus/blob/main/docs/guides/fields.md?

**Reply by Mahjouba91** [reply]
Ok so it's already there, it seems clear enough I think, I did not paid enough attention

**Comment by Mahjouba91** [comment]
Actually, I have found kind of a bug, but it's because of how the current upload image workflow works.

- Enable System collections on the data model options (right sidebar)
- Add a mandatory text field to the Files collection
- In a new entry of any collection with an image field, upload a new image to it, you will get an error message saying the new field you just added is mandatory whereas you had no option to actually fill it.
- This is because the image metadata is not opened directly after a successful upload, it's just a user action wether he wants to edit image metadata or not.

I am not sure if it's really an issue or not, maybe it does not worth updating the whole workflow to fix that small edge case.

<img width="654" alt="Screenshot 2021-06-21 at 16 59 58" src="https://user-images.githubusercontent.com/8670083/122793986-ae4a0e80-d2b3-11eb-80df-e5836473bd03.png">

**Reply by rijkvanzanten** [reply]
Yeah that's gonna require overhauling every place where you can upload an image, and it means you can never upload a file through drag and drop, as you always need an extra step to enter the required fields 🤔

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
